io/src/lib.rs

```diff
@@ -230,6 +230,57 @@ impl<T: AsRef<[u8]>> BufRead for Cursor<T> {
     }
 }
 
+/// A [`Read`]er which keeps an internal buffer to avoid hitting the underlying stream directly for
+/// every read, implementing [`BufRead`].
+pub struct BufReader<'a, R: Read> {
+    inner: &'a mut R,
+    buf: [u8; 4096],
```

Comment:
Maybe the buffer should be const generic.

Reply:
Maybe a generic with default is possible? But if it is, it can also be added later so I'm not going to block the PR over this.